designs/capi-resources-custom-namespace.md

```diff
@@ -0,0 +1,54 @@
+# CAPI Resources in Custom Namespace
+
+## Introduction
+
+**Problem:** Customers managing large scale EKS-A clusters (hundreds or thousands of workload clusters for a given management cluster) expect to have difficulty managing all these resources if they are all in the same namespace 
+
+This document aims to solve that, by allowing customers to specify a new flag on the cluster spec. If set, the underlying CAPI/CAPX/etcd resources will be created under the Cluster's namespace, rather than all dumped into `eksa-system`. 
```

Comment:
so, in this proposal, the user would set a bool flag and EKS A would then create the objects in a namespace named for the cluster, is that right? what if we have multiple workload cluster with the same name (not sure if that's supported anyway but just wondering)?

Reply:
I think we'd run into a problem generating the filesystem folder to store the artifacts for that, right? It seems like we should prevent users from having clusters of the same name, even if they are in different namespaces. It would be an easy preflight check, if that makes sense to you? @danbudris